src/test/java/com/github/checkstyle/regression/git/DiffParserTest.java

```diff
@@ -23,45 +23,106 @@
+            GitUtils.createNewBranchAndCheckout(repository, "foo");
+            GitUtils.addAnEmptyFileAndCommit(repository, "AddedFile");
+            final List<GitChange> changes = DiffParser.parse(
+                    repository.getDirectory().getParent(), "foo");
+            assertEquals(1, changes.size());
+            assertChange("AddedFile", changes.get(0));
+            FileUtils.deleteDirectory(repository.getDirectory().getParentFile());
+        }
     }
 
-    @After
-    public void tearDown() throws Exception {
-        repository.close();
+    @Test
+    public void testParseModifyChange() throws Exception {
+        try (Repository repository = GitUtils.createNewRepository()) {
+            final File helloWorld = GitUtils.addAnEmptyFileAndCommit(repository, "HelloWorld");
+            GitUtils.createNewBranchAndCheckout(repository, "foo");
+            Files.write(helloWorld.toPath(), "hello world!".getBytes(Charset.forName("UTF-8")),
+                    StandardOpenOption.APPEND);
+            GitUtils.addAllAndCommit(repository, "append text to HelloWorld");
+            final List<GitChange> changes = DiffParser.parse(
+                    repository.getDirectory().getParent(), "foo");
+            assertEquals(1, changes.size());
+            assertChange("HelloWorld", changes.get(0));
+            FileUtils.deleteDirectory(repository.getDirectory().getParentFile());
+        }
     }
 
     @Test
-    public void testIsProperUtilsClass() throws Exception {
-        assertUtilsClassHasPrivateConstructor(DiffParser.class);
+    public void testParseDeleteChange() throws Exception {
+        try (Repository repository = GitUtils.createNewRepository()) {
+            GitUtils.addAnEmptyFileAndCommit(repository, "HelloWorld");
+            GitUtils.addAnEmptyFileAndCommit(repository, "FileToDelete");
+            GitUtils.createNewBranchAndCheckout(repository, "foo");
+            GitUtils.removeFileAndCommit(repository, "FileToDelete");
+            final List<GitChange> changes = DiffParser.parse(
+                    repository.getDirectory().getParent(), "foo");
+            assertEquals(1, changes.size());
+            assertChange("/dev/null", changes.get(0));
+            FileUtils.deleteDirectory(repository.getDirectory().getParentFile());
+        }
+    }
+
+    @Test
+    public void testParseRenameChange() throws Exception {
+        try (Repository repository = GitUtils.createNewRepository()) {
+            GitUtils.addAnEmptyFileAndCommit(repository, "HelloWorld");
+            GitUtils.createNewBranchAndCheckout(repository, "foo");
+            GitUtils.addAnEmptyFileAndCommit(repository, "HelloWorldFoo");
+            GitUtils.removeFileAndCommit(repository, "HelloWorld");
+            final List<GitChange> changes = DiffParser.parse(
+                    repository.getDirectory().getParent(), "foo");
+            assertEquals(2, changes.size());
```

Comment:
Should we detect RENAME change? In current production code we would only have ADD, MODIFY and DELETE. And "one RENAME" would be seen as "one DELETE + one ADD". If we turn on "detectRename", then there would be only one change with type RENAME.